docs/source/serving/https.rst

```diff
@@ -0,0 +1,74 @@
+.. _https:
+
+HTTPS Encryption
+================
+
+SkyServe enables secure serving of models over HTTPS, which is essential for handling sensitive data or for models that require secure communication with other services. Currently, SkyServe supports HTTPS only on the load balancer (for communication between the client and the load balancer); HTTPS on the service replicas (for communication between the load balancer and the service replicas) is not yet supported.
+
+HTTPS on Load Balancer
+----------------------
+
+To enable HTTPS on the load balancer, you need to provide a certificate and a private key. Obtaining these from a trusted Certificate Authority (CA) is the most secure method. However, for development and testing purposes, you can generate a self-signed certificate and private key using the :code:`openssl` command-line tool. Here is an example of how to generate them:
+
+.. code-block:: bash
+
+  $ openssl req -x509 -newkey rsa:2048 -days 36500 -nodes -keyout <key-path> -out <cert-path>
+
+You can provide these as files:
+
+.. code-block:: yaml
+  :emphasize-lines: 5-6
+
+  # https.yaml
+  service:
+    readiness_probe: /
+    tls:
+      keyfile: <key-path>
+      certfile: <cert-path>
+
+  resources:
+    ports: 8080
+
+  run: python3 -m http.server 8080
+
+To deploy the service, run the following command:
+
+.. code-block:: bash
+
+  $ sky serve up https.yaml -n https
```

Comment:
maybe name it `my-service` instead? A bit confusing when it is used in `--endpoint https` below.